File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int current = nums[0];
        int indexToPlace=1;
        int k = 1;
        for (int i=1; i<nums.length; i++) {
            if (current!=nums[i]) {
                current=nums[i];
                nums[indexToPlace]=current;
                indexToPlace++;
                k++;
            }
        }
        return k;
}}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = 0
        cur = 0
        prefix_sum = {0:1}

        for num in nums:
            cur += num
            diff = cur - k

            res += prefix_sum.get(diff, 0)
            prefix_sum[cur] = 1 + prefix_sum.get(cur, 0)

        return res